connect.go

```diff
@@ -0,0 +1,471 @@
+package main
+
+import (
+	"fmt"
+	"image"
+	"image/color"
+	"log"
+	"math/rand"
+	"time"
+
+	"github.com/hajimehoshi/ebiten/v2"
+	"github.com/hajimehoshi/ebiten/v2/inpututil"
+	"github.com/hajimehoshi/ebiten/v2/text"
+	"golang.org/x/image/font/basicfont"
+)
+
+const (
+	connectScreenWidth  = 800
+	connectScreenHeight = 600
+	boardWidth         = 16
+	boardHeight        = 10
+	tileSize           = 32
+	boardOffsetX       = (connectScreenWidth - boardWidth*tileSize) / 2
+	boardOffsetY       = (connectScreenHeight - boardHeight*tileSize) / 2
+	tileTypes          = 8
+)
+
+// TileType represents different tile images
+type TileType int
+
+const (
+	TileEmpty TileType = iota
+	TileAnimal1 // Cat
+	TileAnimal2 // Dog
+	TileAnimal3 // Panda
+	TileAnimal4 // Tiger
+	TileFruit1  // Apple
+	TileFruit2  // Orange
+	TileFruit3  // Strawberry
+	TileFruit4  // Grape
+)
+
+var tileColors = map[TileType]color.Color{
+	TileEmpty:   color.RGBA{50, 50, 50, 255},
+	TileAnimal1: color.RGBA{255, 192, 203, 255}, // Pink (Cat)
+	TileAnimal2: color.RGBA{139, 69, 19, 255},   // Brown (Dog)
+	TileAnimal3: color.RGBA{0, 0, 0, 255},       // Black (Panda)
+	TileAnimal4: color.RGBA{255, 140, 0, 255},   // Orange (Tiger)
+	TileFruit1:  color.RGBA{255, 0, 0, 255},     // Red (Apple)
+	TileFruit2:  color.RGBA{255, 165, 0, 255},   // Orange
+	TileFruit3:  color.RGBA{255, 20, 147, 255},  // Deep Pink (Strawberry)
+	TileFruit4:  color.RGBA{128, 0, 128, 255},   // Purple (Grape)
+}
+
+// Tile represents a single tile on the board
+type Tile struct {
+	Type     TileType
+	X, Y     int
+	Selected bool
+	Visible  bool
+}
+
+// ConnectGame represents the main connect game state
+type ConnectGame struct {
+	board         [][]*Tile
+	selectedTile  *Tile
+	score         int
+	gameOver      bool
+	gameWon       bool
+	connectionPath []image.Point
+	showPath      bool
+	pathTimer     float64
+}
+
+// NewConnectGame creates a new connect game instance
+func NewConnectGame() *ConnectGame {
+	g := &ConnectGame{}
+	g.initializeBoard()
+	return g
+}
+
+func (g *ConnectGame) initializeBoard() {
+	g.board = make([][]*Tile, boardHeight+2) // Add border
+	for y := 0; y < boardHeight+2; y++ {
+		g.board[y] = make([]*Tile, boardWidth+2) // Add border
+		for x := 0; x < boardWidth+2; x++ {
+			g.board[y][x] = &Tile{
+				Type:    TileEmpty,
+				X:       x,
+				Y:       y,
+				Visible: false,
+			}
+		}
+	}
+
+	// Generate pairs of tiles
+	tileCount := (boardWidth * boardHeight) / 2
+	if tileCount%2 != 0 {
+		tileCount-- // Ensure even number
+	}
+
+	tiles := make([]TileType, 0, tileCount*2)
+	for i := 0; i < tileCount; i++ {
+		tileType := TileType((i % tileTypes) + 1)
+		tiles = append(tiles, tileType, tileType) // Add pair
+	}
```

Comment:
Current tile generation logic creates pairs but doesn't ensure even distribution of tile types, which can lead to unsolvable boards. The improved version distributes tiles evenly across all types first, then fills remaining slots, ensuring better game balance and solvability.

```suggestion
	// Generate pairs of tiles - ensure even distribution
	tileCount := (boardWidth * boardHeight) / 2
	if tileCount%2 != 0 {
		tileCount-- // Ensure even number
	}

	tiles := make([]TileType, 0, tileCount*2)
	// Generate pairs ensuring each tile type appears an even number of times
	tilesPerType := (tileCount * 2) / tileTypes
	for i := 1; i <= tileTypes; i++ {
		tileType := TileType(i)
		for j := 0; j < tilesPerType; j++ {
			tiles = append(tiles, tileType)
		}
	}
	// Fill remaining slots if any
	for len(tiles) < tileCount*2 {
		tiles = append(tiles, TileType((len(tiles)%tileTypes)+1))
	}
```